### ISSUE TITLE
How to apply splice with seamless-immutable?

### ISSUE THREAD

**Issue by etairi** [rtfeldman/seamless-immutable#251]
I'm using seamless-immutable inside React app, and I have a code segment like this in my saga:

```js
let parsedData = yield select(state => state.post.posts);
parsedData = postUtils.deleteLike(parsedData, payload);
```

And my `deleteLike` method looks like this:

```js
const deleteLike = (parsedPosts, likeId) => {
    let posts = parsedPosts.asMutable();

    Object.entries(posts).forEach(
        ([_, userPosts]) => {
            Object.entries(userPosts).forEach(
                ([_, post]) => {
                    const found = post.likes.filter(l => l.id === likeId);
                    if (found.length > 0) {
                        const index = post.likes.indexOf(found[0]);
                        post.likes.splice(index, 1);
                    }
                }
            );
        }
    );

    return posts;
};

So, basically I want to remove one element from one of the inner arrays that is found inside the object. Though, the above code throws an error saying that splice cannot be invoked on an Immutable data structure. How can I achieve what I want with seamless-immutable?
```
